microcosm/hooks.py

```diff
@@ -77,15 +78,15 @@ def _register_hook(hook_prefix, target_cls, func, *args, **kwargs):
         setattr(target_cls, hook_name, [call])
 
 
-def invoke_resolve_hook(target):
+def invoke_resolve_hook(target: Any) -> None:
     """
     Invoke resolution hook.
 
     """
     return _invoke_hook(ON_RESOLVE, target)
 
 
-def on_resolve(target, func, *args, **kwargs):
+def on_resolve(target: Any, func: Callable[[Any, Any], None], *args, **kwargs) -> None:
```

Comment:
I'm not sure that hooks are `Callable[[Any, Any], None]` – it looks like they are invoked with the target component, *args and **kwargs.  `Callable[..., None]` is probably the best we can do right now.